autobotAI_integrations/utils/__init__.py

```
@@ -391,20 +391,26 @@ def get_header_params(headers):
         )
     return params
 
-def extract(resources: List[Any], selector: str, logger) -> List[Dict[str, Any]]:
+def extract(resources: Union[List[Any], Dict[str, Any]], selector: str, logger) -> List[Dict[str, Any]]:
```

Ruff (0.15.13)

[warning] Too many branches (13 > 12)

(PLR0912)